### PR TITLE
feat: set_default for Animation class

### PR DESCRIPTION
**Pull request by christopher-hampson**
## Overview: What does this pull request change?

Adds set_default class method to Animation class based on existing implementation for Mobject class.

## Motivation and Explanation: Why and how do your changes improve the library?

Addresses issue [#3142](https://github.com/ManimCommunity/manim/issues/3142)

## Reviewer Checklist
- [ ] The PR title is descriptive enough for the changelog, and the PR is labeled correctly
- [ ] If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
- [ ] If applicable: newly added functions and classes are tested

### CODE REVIEW

**Comment by JasonGrace2282** [comment]
Hello, thanks for sending a PR for this feature :)
Off of a quick look, the implementation seems fine (I'll take a closer look, hopefully later today). Would you mind adding some tests for this behavior?
Thanks for helping make Manim better :sparkle:

P.S. Don't worry about the pre-commit failure. We'll fix that in a different PR.

File: manim/animation/animation.py [hunk 1]
```diff
@@ -479,6 +480,52 @@ def is_introducer(self) -> bool:
...
+                    Rotate.set_default(run_time=2,rate_func=rate_functions.linear)
+                    Indicate.set_default(color=None)
+
+                    S = Square(color=BLUE,fill_color=BLUE,fill_opacity=0.25)
+                    self.add(S)
+                    self.play(Rotate(S,PI))
+                    self.play(Indicate(S))
+
+                    Rotate.set_default()
+                    Indicate.set_default()
```
Comment (JasonGrace2282) [inline manim/animation/animation.py:521]: Thanks for writing an example (especially one that renders fast)!

**Review by JasonGrace2282 — CHANGES_REQUESTED** [review]
Just tested and it works for me! A simple test to add would be something like
```py
def test_animation_set_default():
    s = Square()
    Rotate.set_default(run_time=100)
    anim = Rotate(s)
    assert anim.run_time == 100
```
Other than that, looks good to me :)

**Comment by christopher-hampson** [comment]
> Just tested and it works for me! A simple test to add would be something like
> 
> ```python
> def test_animation_set_default():
>     s = Square()
>     Rotate.set_default(run_time=100)
>     anim = Rotate(s)
>     assert anim.run_time == 100
> ```
> 
> Other than that, looks good to me :)

Thanks for reviewing and suggesting this test! I'll get this added shortly. :+1:

File: manim/animation/animation.py [hunk 2]
```diff
@@ -479,6 +480,52 @@ def is_introducer(self) -> bool:
...
+            function of this class.
+
+        Examples
+        --------
+
+        .. manim:: ChangeDefaultAnimation
+
+            class ChangeDefaultAnimation(Scene):
+                def construct(self):
+                    Rotate.set_default(run_time=2,rate_func=rate_functions.linear)
```
Comment (behackl) [inline manim/animation/animation.py:512]:
```suggestion
                    Rotate.set_default(run_time=2, rate_func=rate_functions.linear)
```

File: manim/animation/animation.py [hunk 3]
```diff
@@ -479,6 +480,52 @@ def is_introducer(self) -> bool:
...
+                    Indicate.set_default(color=None)
+
+                    S = Square(color=BLUE,fill_color=BLUE,fill_opacity=0.25)
```
Comment (behackl) [inline manim/animation/animation.py:515]:
```suggestion
                    S = Square(color=BLUE, fill_color=BLUE, fill_opacity=0.25)
```

File: manim/animation/animation.py [hunk 4]
```diff
@@ -479,6 +480,52 @@ def is_introducer(self) -> bool:
...
+                    self.add(S)
+                    self.play(Rotate(S,PI))
```
Comment (behackl) [inline manim/animation/animation.py:517]:
```suggestion
                    self.play(Rotate(S, PI))
```

**Review by JasonGrace2282 — APPROVED** [review]
LGTM, thanks :)

**Pull request merged**